packages/account/src/Components/currency-selector/currency-selector.tsx

```diff
@@ -1,294 +1,309 @@
+                onSubmitEnabledChange?.(!is_submit_disabled);
             }
-            resetRealAccountSignupParams();
-            setIsBypassStep(false);
-        }
-        // eslint-disable-next-line react-hooks/exhaustive-deps
-    }, [is_bypass_step]);
+        };
 
-    const getHeightOffset = () => {
-        if (is_appstore) {
-            return '222px';
-        } else if (!has_currency && has_real_account) {
-            return '89px';
-        }
-        return '159px';
-    };
+        const handleCancel = (values: TCurrencySelectorFormProps) => {
+            const current_step = getCurrentStep() - 1;
+            onSave(current_step, values);
+            onCancel(current_step, goToPreviousStep);
+        };
 
-    const getSubmitLabel = () => {
-        if (set_currency) {
-            return localize('Set currency');
-        } else if (has_wallet_account) {
-            return localize('Finish');
-        }
-        return localize('Next');
-    };
+        const handleValidate = (values: TCurrencySelectorFormProps) => {
+            checkSubmitStatus(values);
+            const { errors } = splitValidationResultTypes(validate(values));
+            return errors;
+        };
 
-    const description = React.useMemo(() => {
-        const dmt5_label = is_eu ? localize('CFDs') : localize('Deriv MT5');
-        const platform_name_dxtrade = getPlatformSettings('dxtrade').name;
+        // In case of form error bypass to update personal data
+        React.useEffect(() => {
+            if (real_account_signup?.error_code) {
+                setIsBypassStep(true);
+            }
+        }, [real_account_signup?.error_code]);
+
+        React.useEffect(() => {
+            if (is_bypass_step && real_account_signup?.error_details) {
+                const keys = Object.keys(real_account_signup?.error_details);
+                const route_to_address_details = Object.keys(getAddressDetailsFields()).filter(item =>
+                    keys.includes(item)
+                );
+                if (route_to_address_details?.length > 0) goToStep(3);
+                else {
+                    goToNextStep();
+                }
+                resetRealAccountSignupParams();
+                setIsBypassStep(false);
+            }
+            // eslint-disable-next-line react-hooks/exhaustive-deps
+        }, [is_bypass_step]);
+
+        const getHeightOffset = () => {
+            if (is_appstore) {
+                return '222px';
+            } else if (!has_currency && has_real_account) {
+                return '89px';
+            }
+            return '159px';
+        };
+
+        const getSubmitLabel = () => {
+            if (set_currency) {
+                return localize('Set currency');
+            } else if (has_wallet_account) {
+                return localize('Finish');
+            }
+            return localize('Next');
+        };
+
+        const description = React.useMemo(() => {
```

Comment:
there is quite simple logic and small code, do we really need here useMemo?
Considering, that radio button group component is not wrapped with React.memo